lib/rubocop/cop/lint/useless_rescue.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for useless `rescue`s, which only reraise rescued exceptions.
+      #
+      # @example
+      #   # bad
+      #   def foo
+      #     do_something
+      #   rescue
+      #     raise
+      #   end
+      #
+      #   # good
+      #   def foo
+      #     do_something
+      #   rescue
+      #     do_cleanup
+      #     raise
+      #   end
+      #
+      #   # bad (latest rescue)
+      #   def foo
+      #     do_something
+      #   rescue ArgumentError
+      #     # noop
+      #   rescue
+      #     raise
+      #   end
+      #
+      #   # good (not the latest rescue)
+      #   def foo
+      #     do_something
+      #   rescue ArgumentError
+      #     raise
+      #   rescue
+      #     # noop
+      #   end
+      #
+      class UselessRescue < Base
+        MSG = 'Useless `rescue` detected.'
+
+        def on_rescue(node)
+          resbody_node = node.resbody_branches.last
+          add_offense(resbody_node) if only_reraising?(resbody_node)
+        end
+
+        private
+
+        def only_reraising?(resbody_node)
+          body = resbody_node.body
+          return false if body.nil? || !body.send_type? || !body.method?(:raise)
+
+          !body.arguments? ||
+            (body.arguments.size == 1 &&
+              body.first_argument.source == resbody_node.exception_variable&.source)
```

Comment:
Should this also handle things like:
```ruby
rescue
  raise $!
```

```ruby
rescue
  raise $ERROR_INFO
```

It's kinda dumb, but I've seen devs do this.

Reply:
Thanks, updated with these suggestions.

Reply:
You should also mention them in the doc examples.

Reply:
Added a mention.